### PR TITLE
Core 2557 typeerror

### PR DESCRIPTION
**Pull request by gmccue**
Fixes the following errors:
* `work dir does not exist`
* `TypeError: Cannot convert undefined or null to object`
* `Cannot read property 'browserType' of undefined`
* Downloads GeoIP city DB from s3 since existing link no longer works

### CODE REVIEW

File: features.js [hunk 1]
```diff
@@ -556,7 +556,7 @@ module.exports = {
     // TODO: edge, mobile platforms?
     browserType: function(client, peerConnectionLog) {
         var peerConnectionConfig = getPeerConnectionConfig(peerConnectionLog);
-        return peerConnectionConfig.browserType || 'unknown';
+        return 'browserType' in peerConnectionConfig ? peerConnectionConfig.browserType : 'unknown';
```
Comment (cmrigney) [inline features.js]: Why was this change needed?

File: app.js [hunk 2]
```diff
@@ -108,14 +108,22 @@ var q = new ProcessQueue();
...
-            fs.unlinkSync(tempPath + '/' + fname);
-        });
-        fs.rmdirSync(tempPath);
-    } catch(e) {
-        console.error('work dir does not exist');
+        if (fs.existsSync(tempPath)) {
+            fs.readdirSync(tempPath).forEach(fname => {
+                try {
+                    console.log('Removing file ' + tempPath + '/' + fname);
+                    fs.unlinkSync(tempPath + '/' + fname);
```
Comment (cmrigney) [inline app.js:115]: In the context of Node, it's preferable to use `path.join` instead since it accounts for OS differences and handling slashes.

File: app.js [hunk 3]
```diff
@@ -108,14 +108,22 @@ var q = new ProcessQueue();
...
-        fs.readdirSync(tempPath).forEach(fname => {
-            fs.unlinkSync(tempPath + '/' + fname);
-        });
-        fs.rmdirSync(tempPath);
-    } catch(e) {
-        console.error('work dir does not exist');
+        if (fs.existsSync(tempPath)) {
+            fs.readdirSync(tempPath).forEach(fname => {
+                try {
+                    console.log('Removing file ' + tempPath + '/' + fname);
```
Comment (cmrigney) [inline app.js:114]: If this is a newer version of node, you can use the es6 feature template literals. Eg `<backtick>Removing file ${tempPath}/${fname}<backtick>`

File: app.js [hunk 4]
```diff
@@ -108,14 +108,22 @@ var q = new ProcessQueue();
...
+                    fs.unlinkSync(tempPath + '/' + fname);
+                } catch (e) {
+                    console.error('Error while unlinking file ' + fname + ' - ' + e.message);
```
Comment (cmrigney) [inline app.js:117]: Template literals here too.

File: app.js [hunk 5]
```diff
@@ -108,14 +108,22 @@ var q = new ProcessQueue();
...
+                }
+            });
+        } else {
+            console.log('Creating working dir ' + tempPath);
```
Comment (cmrigney) [inline app.js:121]: Template literals.

File: app.js [hunk 6]
```diff
@@ -108,14 +108,22 @@ var q = new ProcessQueue();
...
+            fs.mkdirSync(tempPath);
+        }
+    } catch (e) {
+        console.error('Error while accessing working dir ' + tempPath + ' - ' + e.message);
```
Comment (cmrigney) [inline app.js:125]: Template literals.

**Review by cmrigney — COMMENTED** [review]
Some questions/comments.

File: app.js (hunk 2 above)
Comment (gmccue) [reply app.js:115]: Since this is a fork, I opted to use (minus the `logger` library and ES6 template literals) the change introduced into the parent project, so that if we want to merge from upstream later on it would be easier: https://github.com/fippo/rtcstats-server/blob/master/app.js#L112

I could change this though. Any thoughts?

File: features.js (hunk 1 above)
Comment (gmccue) [reply features.js]: This is to fix the `undefined property error` as shown here: https://app.datadoghq.com/logs?cols=service&event=AQAAAW_tEXEAqPN0PAAAAABBV190RVhFQXhNaDFWQ2dNTXV0cA&from_ts=1580229807720&index=main&live=true&messageDisplay=inline&query=browserType&stream_sort=desc&to_ts=1580230707720

File: app.js (hunk 3 above)
Comment (gmccue) [reply app.js:114]: I opted to keep the legacy format here as the rest of the codebase does not use template literals. If we wanted to move to template literals it might be best done in a refactor ticket? Any thoughts?

Comment (cmrigney) [reply app.js:114]: Yeah makes sense. It probably depends on this codebase. If it's something we're maintaining, then usually we refactor as we make changes. But since this is a fork of another project, we may want to be careful not to get too different for the sake of merges from the original.

File: app.js (hunk 2 above)
Comment (cmrigney) [reply app.js:115]: Ok, in that case that's fine.

File: features.js (hunk 1 above)
Comment (cmrigney) [reply features.js]: Ah, in that case I believe this will still throw an undefined error since `in` requires that peerConnectionConfig be defined. And the error is showing that peerConnectionConfig is undefined.  You could do something like `return (peerConnectionConfig && peerConnectionConfig.browserType) || 'unknown';`

Comment (gmccue) [reply features.js]: Ah yeah, good catch. Fixed.

**Review by cmrigney — APPROVED** [review]
lgtm
